src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -176,7 +176,7 @@ public function searchBarAction(Request $request): Response
             $this->cspHandler->disableCsp();
         }
 
-        if (!$request->hasSession()) {
+        if (true === $request->attributes->getBoolean('_stateless') || !$request->hasSession()) {
```

Comment:
Actually, this code looks broken to me. When we don't have a session, we should still support reading the query string.